Assets/Scripts/Entity.cs

```diff
@@ -0,0 +1,82 @@
+using System.Collections;
+using UnityEngine;
+
+public class Entity : MonoBehaviour, IDamagable
+{
+    [SerializeField]
+    public float initialHealth = 100;
+    public float health { get { return _health; } set { _health = value; } }
+    [SerializeField]
+    private float _health = 100;
+
+    [SerializeField]
+    private float destroyDelay = 1.5f;
+
+    public bool isDead { get => health <= 0 && _dead; }
+    private bool _dead = false;
+
+    // component references
+    private Accelerometer accelerometer;
+
+    private void Start()
+    {
+        _health = initialHealth;
+        accelerometer = GetComponent<Accelerometer>();
+        if (accelerometer == null)
+        {
+            accelerometer = gameObject.AddComponent(typeof(Accelerometer)) as Accelerometer;
+        }
+    }
+
+    private void Awake()
+    {
+        
+    }
+
+    private void Update()
+    {
+        if (health <= 0)
+        {
+            Die();
+        }
+    }
+
+    public void OnCollisionEnter2D(Collision2D collision)
+    {
+        Entity otherEntity = collision.gameObject.GetComponent<Entity>();
+        Accelerometer otherAccelerometer = collision.gameObject.GetComponent<Accelerometer>();
+        if (otherAccelerometer)
+        {
+            TakeDamage(otherAccelerometer.impactForce.magnitude);
+            if(otherEntity)
+            {
+                otherEntity.TakeDamage(accelerometer.impactForce.magnitude);
+            }
+        }
+        TakeDamage(accelerometer.impactForce.magnitude);
+    }
+
+    public void TakeDamage(float damage)
+    {
+        health -= damage;
+    }
+
+    public void Die()
+    {
+        if (!_dead)
+        {
+            health = 0;
+            DeathRoutine();
+            StartCoroutine(DeactivateAfterDelay());
+            _dead = true;
+        }
+    }
+
+    public virtual void DeathRoutine(){}
+
+    private IEnumerator DeactivateAfterDelay()
```

Comment:
I think this should be called as the default DeathRoutine and DeactivateAfterDelay should accept the parameter for the delay. If a child class overrides the DeathRoutine it is their responsibility to opt into calling DeactiveAfterDealy(time) now.

Reply:
Done: https://github.com/cjh15hub/terribletweeters/pull/1/commits/81f82db7e00559acff3871b77076b9bdb9bcaa43#diff-8446b20d766823004b26f3f1b33f5831cf745c6d4a21828ccea0479ff1c54ce9R92